### PR TITLE
Implement a prereqCmd option for installing system deps

### LINKED ISSUE #101: Need to install Python module that have system dependencies
## Problem
Unable to install Python modules that have system dependencies, such as mysqlclient, and there's no faculty in this package to do so.

## Suggested Solution
Add support for an optional pre-requirements shell script to be run.  This shell script (which the end-user must write) must be able to both install system packages, and then copy any desired artifacts into the folder to get packaged.  In my above instance, I would suggest something like...

```
  pythonRequirements:
    dockerizePip: true
    preReqScript: install-mysql-devel.sh
```
and that shell script `install-mysql-devel.sh` (in the same folder as serverless.yml) would contain...
```
#!/bin/bash
yum install -y mysql-devel
cp /usr/lib64/mysql/libmysqlclient.so.18.0.0 ./libmysqlclient.so.18
```

I'll put a bounty on this feature of ~$100 either in money or stuff, your choice.  I need this badly, and don't have the time to make this feature right now.  I will eventually implement this if no one's free because I'll need it and it will annoy me too much.

### PR DESCRIPTION
**Pull request by dschep**
closes #101 

Test this out and let me know if it fulfills your requirements @AndrewFarley

(install with `npm i UnitedIncome/serverless-python-requirements@prereq-script`)

### CODE REVIEW

**Comment by dschep** [comment]
However! Another approach I have considered in the past is building a serverless plugin just for installing system dependencies. Thoughts on that approach? (you'd just add `serverless-aws-system-deps` to your plugins and add an option under `custom` like: `deps: [mysql-dev]`. The only issue was that I wasn't sure what to grab or where to copy things after the system package is installed.

**Comment by AndrewFarley** [comment]
@dschep Glanced at the files changed, looks great.  Will try it out this evening and let you know.  Cheers!

**Comment by AndrewFarley** [comment]
Hey @dschep a quick check of this and it doesn't seem to run prereqCmd at all for me.
```
  pythonRequirements:
    dockerizePip: true
    invalidateCaches: false
    prereqCmd: bash -c "yum install mysql-devel && cp /usr/lib64/mysql/libmysqlclient.so.18.0.0 ./libmysqlclient.so.18"
    #    prereqCmd: ./install-mysql-client.sh

```
I tried both ways, inline shell and included file.  I dropped some debug echoes into the prereqCmd.js and that function never gets executed.  Am I missing something?

**Comment by dschep** [comment]
Hmm.. When you did an included file, how did you specify the name/path? In my test I used `./foobar` (a relative path). I think that it has to be a path that node's `child_process.Spawn` can excecute. I'll add the `shell` option which might make the bash command work. I'll add a test for that too.

**Comment by AndrewFarley** [comment]
Well... the prereq command isn't even running at all for me, as I said I threw some quick debug console logs in there to show it.  A good example to test against, it making sure that requirements.txt has "mysqlclient" in it, and that pip package compiles properly.

**Comment by dschep** [comment]
try it now. it looks like the `shell` change should do the trick.

**Comment by dschep** [comment]
did  you try again with the new commits?

**Comment by AndrewFarley** [comment]
Yeah, still no go.  I don't know if it's because I'm on OS-X, or perhaps using it dockerized?

**Comment by dschep** [comment]
mmm yeah, i'll have to test it dockerized.

**Comment by AndrewFarley** [comment]
Yeah, sorry @dschep the whole point of this was for use while in dockerized pip mode.  Otherwise, I would/could have just done some simple wrapper to install and some local system libraries.  :)  Hoping this can still happen one day soon though.

**Comment by Arsen00** [comment]
open

File: README.md [hunk 1]
```diff
@@ -113,6 +113,16 @@ custom:
     pythonBin: /opt/python3.6/bin/python
 ```
 
+### Run Custom extra packaging scripts
```
Comment (Arsen00) [inline README.md]: +### Run Custom extra packaging scripts

**Comment by dschep** [comment]
@Arsen00, What?

**Pull request closed without merging**